include/picongpu/simulationControl/MySimulation.hpp

```diff
@@ -803,6 +791,25 @@ class MySimulation : public SimulationHelper<simDim>
     int32_t endSlidingOnStep;
     bool showVersionOnce;
     bool autoAdjustGrid = true;
+
+private:
+
+    void initFields( DataConnector& dataConnector )
+    {
+        using pmacc::memory::makeUnique;
+        auto fieldB = makeUnique< FieldB >( *cellDescription );
+        dataConnector.consume( std::move( fieldB ) );
```

Comment:
Note: this `move` is required, since `fieldB` is an lvalue. If one puts `makeUnique< FieldB >( *cellDescription )` directly instead of `fieldB `, that would work without the move. However, I feel readability is more important here.